digicampipe/image/utils.py

```diff
@@ -0,0 +1,416 @@
+from astropy.io import fits
+from astropy.stats import sigma_clipped_stats
+import numpy as np
+import scipy
+from skimage.draw import polygon
+import matplotlib.pyplot as plt
+from time import clock
+import cv2
+from scipy import signal, ndimage, optimize
+import numbers
+from photutils import DAOStarFinder
+
+
+def crop_image(image, crop_pixel1, crop_pixel2):
+    """
+    :param image: 2D array of the image data with a shape = (vertical size, horizontal size)
+    :param crop_pixel1: iterable of 2 ints: position (pos_x , pos_y) of one corner of the returned region
+    :param crop_pixel2: iterable of 2 ints: position (pos_x , pos_y) of the opposite corner of the returned region
+    :return: the cropped image, the 1st corner and the opposite corner positions
+    """
+    if type(image) is str:
+        image = fits.open(image)[0].data
+    if type(image) is not np.ndarray:
+        raise AttributeError("image must be a filename or a numpy array.")
```

Comment:
I would not do this type check here, later this function does stuff with image, which is only possible if it is a 2d-ndarray. that is the time to raise the exception.... not now.